subsys/bluetooth/controller/CMakeLists.txt

```diff
@@ -187,6 +187,11 @@ zephyr_library_include_directories_ifdef(
   hci/nordic
   )
 
+zephyr_library_sources_ifdef(
+  CONFIG_SOC_COMPATIBLE_NRF
+  hci/nordic/hci_vendor.c
+  )
+
```

Comment:
Thank you for discovering.

1. Add it here: https://github.com/cvinayak/zephyr/blob/2de71fa1d7eb3e5b29b52dd0be9d1f6eb8b52572/subsys/bluetooth/controller/ll_sw/nordic/CMakeLists.txt#L87 and respectively for `openisa` too.

as:
```c
zephyr_library_sources(
  hci/hci_vendor.c
  )
```

2. Move the vendor specific {nordic, openisa} header and source files too into `ll_sw/{nordic,openisa}/hci/` 

(maybe easy said by me, but give a try else I can do it).

Reply:
okay, had to adjust a path as well, did I get this right?